pkg/controllers/capr/plansecret/plansecret.go

```diff
@@ -103,7 +106,9 @@ func (h *handler) OnChange(key string, secret *corev1.Secret) (*corev1.Secret, e
 
 	if failedChecksum == planner.PlanHash(plan) {
```

Comment:
Can you add a comment here detailed that we will set the failed checksum, but not mark the plan as truly failed unless it is either 1. not allowed to fail or 2. has reached the max failure count